api/src/main/resources/api/service_openapi.yaml

```diff
@@ -439,6 +515,16 @@ components:
         nextPageToken:
           $ref: '#/components/schemas/NextPageToken'
 
+    Relationship:
+      type: object
+      description: A relationship to another entity.
+      properties:
+        name:
+          type: string
+          description: The name of the relationship.
+        relatedEntity:
+          $ref: '#/components/schemas/Entity'
```

Comment:
`entity.relationships[].relatedEntity.relationships[].relatedEntity...`

I would probably just use the entity name here to avoid these kinds of issues and potential cycles

Reply:
SG, does seem simpler. Will also change the proto. But it does mean you may need to call this end point multiple times during construction.